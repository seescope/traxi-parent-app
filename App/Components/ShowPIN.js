import React from 'react';
import { View, Text } from 'react-native';
import Intercom from 'react-native-intercom';

import HeaderText from './HeaderText';
import Spacing from './Spacing';
import Button from './Button';
import LoadingIndicator from './LoadingIndicator';
import { WHITE } from '../Constants/Colours';

const PINstyle = {
  fontSize: 64,
  color: WHITE,
  textAlign: 'center',
};

const buttonContainer = {
  flex: 1,
  alignItems: 'center',
};

const ShowPIN = ({ setupID, kidName }) => (
  <View>
    <HeaderText>
      On {kidName}'s device {'\n'}
      go to mytraxi.com {'\n'}
      and enter the code:
    </HeaderText>

    <Spacing height={64} />

    <Text style={PINstyle}>
      {setupID}
    </Text>

    <Spacing height={64} />

    <LoadingIndicator>Waiting for {kidName}'s device...</LoadingIndicator>

    <Spacing height={32} />

    <View style={buttonContainer}>
      <Button primary={false} onPress={() => Intercom.displayMessageComposer()}>
        I need help
      </Button>
    </View>
  </View>
);

ShowPIN.propTypes = {
  setupID: React.PropTypes.number.isRequired,
  kidName: React.PropTypes.string.isRequired,
};

export default ShowPIN;
