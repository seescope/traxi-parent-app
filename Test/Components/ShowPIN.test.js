import React from 'react';
import renderer from 'react-test-renderer';
import ShowPIN from '../../App/Components/ShowPIN';

it('renders', () => {
  const tree = renderer.create(<ShowPIN setupID={1234} kidName="Test" />).toJSON();
  expect(tree).toMatchSnapshot();
});
