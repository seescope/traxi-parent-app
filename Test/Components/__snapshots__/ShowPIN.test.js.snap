exports[`test renders 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-ExtraBold",
        "fontSize": 30,
        "marginBottom": -2,
        "textAlign": "center",
      }
    }>
    On 
    Test
    \'s device 
    
    
    go to mytraxi.com 
    
    
    and enter the code:
  </Text>
  <View
    style={
      Object {
        "height": 64,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontSize": 64,
        "textAlign": "center",
      }
    }>
    1234
  </Text>
  <View
    style={
      Object {
        "height": 64,
      }
    } />
  <View>
    <ActivityIndicator
      animating={true}
      color="#FFF"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "marginBottom": 16,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          Object {
            "backgroundColor": "transparent",
            "color": "#FFF",
            "textAlign": "center",
          },
        ]
      }>
      Waiting for 
      Test
      \'s device...
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 32,
      }
    } />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "Raleway-Regular",
            "fontSize": 20,
          }
        }>
        I need help
         
      </Text>
    </View>
  </View>
</View>
`;
